src/nagini_translation/analyzer.py

```diff
@@ -460,15 +460,53 @@ def visit_ClassDef(self, node: ast.ClassDef) -> None:
                 if not (isinstance(base, ast.Name) and
                         base.id in MYPY_SUPERCLASSES):
                     actual_bases.append(base)
+
+        if len(actual_bases) > 0:
+            cls.superclass = self.find_or_create_target_class(actual_bases[0])
+
+            # Process ADT defined by a class hierarchy (sum) and a NamedTuple (product)
+            if isinstance(cls.superclass, PythonClass) and cls.superclass.is_adt:
```

Comment:
This seems like ADT constructors always have to name the ADT superclass first and the NamedTuple second. That's fine by me if it makes things easier, but we should make sure we get an okay error message if someone puts them in a different order.

Reply:
A good message seems fair, although Nagini currently does not support NamedTuple as a superclass, generically speaking. Attempting to do so will raise a not supported feature exception "class literal has unsupported format". Since only the first base class is checked (multiple inheritance is not supported), if it is a NamedTuple, verification halts with above exception before we have a chance to detect the intention was to define an ADT. If the first base class is an ADT, though, then the following NamedTuple is processed particularly for ADTs (not generically). I guess this is the best we can do now, since a cleaner solution would be to fully support NamedTuples and multiple inheritance.

Reply:
Okay, sounds good.